python-lib/google_calendar_client.py

```diff
@@ -0,0 +1,55 @@
+import logging
+import datetime
+from googleapiclient.discovery import build
+from google.auth.transport.requests import Request
+from google.oauth2.credentials import Credentials
+
+
+SCOPES = ['https://www.googleapis.com/auth/calendar.readonly']
+
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO,
+                    format='google-calendar plugin %(levelname)s - %(message)s')
+
+
+class GoogleCalendarClientError(ValueError):
+    pass
+
+
+class GoogleCalendarClient():
+    def __init__(self, token):
+        credentials = Credentials(token, SCOPES)
+        if not credentials or not credentials.valid:
+            if credentials and credentials.expired and credentials.refresh_token:
+                credentials.refresh(Request())
+            else:
+                raise GoogleCalendarClientError("Credential not valid or need to be refreshed")
+        self.service = build('calendar', 'v3', credentials=credentials)
+
+    def get_events(self, from_date, to_date=None, calendar_id="primary", records_limit=-1, can_raise=True):
+        if from_date is None:
```

Comment:
This default must be made by the caller.
The client mustn't do any business choice